jira/src/main/kotlin/gropius/sync/jira/JiraDataService.kt

```diff
@@ -1,33 +1,77 @@
 
+import com.fasterxml.jackson.databind.JsonNode
+import com.fasterxml.jackson.databind.ObjectMapper
 import gropius.model.architecture.IMSProject
 import gropius.model.issue.Label
 import gropius.model.template.*
+import gropius.model.user.GropiusUser
+import gropius.model.user.IMSUser
 import gropius.model.user.User
+import gropius.repository.user.GropiusUserRepository
+import gropius.sync.JsonHelper
 import gropius.sync.SyncDataService
-import gropius.sync.user.UserMapper
+import gropius.sync.jira.config.IMSConfig
+import gropius.util.JsonNodeMapper
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.plugins.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.plugins.logging.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import io.ktor.serialization.kotlinx.json.*
 import kotlinx.coroutines.reactive.awaitFirstOrNull
 import kotlinx.coroutines.reactor.awaitSingle
-import kotlinx.serialization.json.JsonElement
-import kotlinx.serialization.json.jsonObject
-import kotlinx.serialization.json.jsonPrimitive
+import kotlinx.coroutines.reactor.awaitSingleOrNull
+import kotlinx.serialization.json.*
+import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Qualifier
 import org.springframework.data.neo4j.core.ReactiveNeo4jOperations
 import org.springframework.stereotype.Component
+import java.net.URI
 import java.time.OffsetDateTime
+import java.util.*
 
 /**
  * Context for Jira Operations
- * @param userMapper UserMapper to map Jira Users to Gropius Users
  * @param neoOperations Neo4jOperations to access the database
+ * @param tokenManager Reference for the spring instance of JiraTokenManager
+ * @param helper Reference for the spring instance of JsonHelper
+ * @param objectMapper Reference for the spring instance of ObjectMapper
+ * @param jsonNodeMapper Reference for the spring instance of JsonNodeMapper
+ * @param gropiusUserRepository Reference for the spring instance of GropiusUserRepository
  */
 @Component
 class JiraDataService(
-    val userMapper: UserMapper,
     @Qualifier("graphglueNeo4jOperations")
-    val neoOperations: ReactiveNeo4jOperations
+    val neoOperations: ReactiveNeo4jOperations,
+    val tokenManager: JiraTokenManager,
+    val helper: JsonHelper,
+    val objectMapper: ObjectMapper,
+    val jsonNodeMapper: JsonNodeMapper,
+    val gropiusUserRepository: GropiusUserRepository
 ) : SyncDataService {
 
+    /**
+     * Logger used to print notifications
+     */
+    final val logger = LoggerFactory.getLogger(JiraDataService::class.java)
+
+    /**
+     * Http Client for Jira API requests
+     */
+    val client = HttpClient() {
+        expectSuccess = true
+        install(Logging)
+        install(ContentNegotiation) {
+            json(Json {
+                ignoreUnknownKeys = true
+            }, contentType = ContentType.parse("application/json; charset=utf-8"))
```

Comment:
`ContentType.Application.Json`?

Reply:
no. that would result in `application/json`, I need `application/json; charset=utf-8`

Reply:
wtf is wrong with Jira

Reply:
but as you use it 3 times, maybe a constant?